### DISCUSSION TITLE
Customizing template plugin update for CKAN 2.8.2 to CKAN 2.9.5

### DISCUSSION THREAD

**Discussion by muhammetcetinyamac** [ckan/ckan#6877] [Q&A]
A template plugin previously written for ckan 2.8.2 version does not work stable in ckan version 2.9.5. Are there any ways to update it for ckan 2.9.5? Or do I need to update the template manually ?

**Comment by amercader** [comment]
You might need to update your template yes. Is there an error raised when you try to use it or it's just not rendered?
If there is an error please paste the stack trace here

**Reply by muhammetcetinyamac** [reply]
Hello there! Thank you for the detailed information. It will be of great use to me.
Actually, the template is being processed and there doesn't seem to be a problem. But there are problems with css settings. For example, I had to make a change like in the image below and it took me about a day to find the solution. How do you solve such problems? Do we have to examine one by one? Thank you.

eg for version 2.9.5 > ckan/ckan/templates/base.html 
![image](https://user-images.githubusercontent.com/57544879/171593419-0320fe90-8d86-46be-9372-2962d7d0ef35.png)

**Comment by duttonw** [comment]
many times moved from _ to .  
My team @ThrawnCA  and I have been having lots of fun finding these easter eggs. see https://github.com/qld-gov-au/ckanext-data-qld/pull/146/files#diff-79ac3597bad18a91da66a455f877f0ea8c79b72f3a2e969dc0f34c852957be4e
and other work in our theme's to accomodate said changes. 

Also if your theme extension overrides any paths and you want to be 2.8 and 2.9 compatible, you need to split pylons and flask config blueprints.  See https://github.com/qld-gov-au/ckanext-data-qld/blob/develop/ckanext/data_qld/plugin.py#L38 

I'm unsure how much of the work we have done has gone upstream on the many plugins our website uses.  https://staging.data.qld.gov.au/api/action/status_show

We are in the mist of going from 2.8.x to 2.9.x in the coming weeks and tried to ensure our theme and plugins work in both. (still on python2 till we are fully on 2.9.x)

**Reply by muhammetcetinyamac** [reply]
Thank you so much. I will review your links.

**Comment by ThrawnCA** [comment]
As well as the changes in route names, there are situations where parameters like `pkg_dict` are defined directly, instead of using the `c` object (eg `c.pkg_dict`), and similar changes. It's often possible to find syntax that will work on both 2.8 and 2.9, eventually.

**Reply by muhammetcetinyamac** [reply]
Thank you for your interest. I fix the route names as I see the wrong places.
